workflow/kf-rnaseqv4-wf.cwl

```diff
@@ -0,0 +1,878 @@
+        valueFrom: |
+          $([self])
+      read_length: rmats_read_length
+      variable_read_length: rmats_variable_read_length
+      read_type: rmats_read_type
+      strandedness:
+        source: wf_strand_param
+        valueFrom: |
+          $(self == "rf-stranded" ? "fr-firststrand" : self == "fr-stranded" ? "fr-secondstrand" : "fr-unstranded")
+      novel_splice_sites: rmats_novel_splice_sites
+      stat_off: rmats_stat_off
+      allow_clipping: rmats_allow_clipping
+      output_basename: output_basename
+      rmats_threads: rmats_threads
+      rmats_ram: rmats_ram
+    out: [filtered_alternative_3_prime_splice_sites_jc, filtered_alternative_5_prime_splice_sites_jc,
+      filtered_mutually_exclusive_exons_jc, filtered_retained_introns_jc, filtered_skipped_exons_jc]
+
+  strand_parse:
+    run: ../tools/expression_parse_strand_param.cwl
+    in:
+      wf_strand_param: wf_strand_param
+    out: [rsem_std, kallisto_std, rnaseqc_std, arriba_std]
+
+  star_fusion_1-10-1:
+    run: ../tools/star_fusion_1.10.1_call.cwl
+    in:
+      Chimeric_junction: star_2-7-10a/chimeric_junctions
+      genome_tar: FusionGenome
+      output_basename: output_basename
+      genome_untar_path: star_fusion_genome_untar_path
+      compress_chimeric_junction: compress_chimeric_junction
+    out: [abridged_coding, chimeric_junction_compressed]
+
+  arriba_fusion_2-2-1:
+    run: ../tools/arriba_fusion_2.2.1.cwl
+    in:
+      genome_aligned_bam:
+        source: [samtools_sort/sorted_bam, samtools_sort/sorted_bai]
+        valueFrom: |
+          ${
+            var bundle = self[0];
+            bundle.secondaryFiles = [self[1]];
+            return bundle;
+          }
+      memory: arriba_memory
+      reference_fasta: reference_fasta
+      gtf_anno: gtf_anno
+      outFileNamePrefix: output_basename
+      arriba_strand_flag: strand_parse/arriba_std
+    out: [arriba_fusions]
+
+  arriba_draw_2-2-1:
+    run: ../tools/arriba_draw_2.2.1.cwl
+    in:
+      fusions: arriba_fusion_2-2-1/arriba_fusions
+      genome_aligned_bam:
+        source: [samtools_sort/sorted_bam, samtools_sort/sorted_bai]
+        valueFrom: |
+          ${
+            var bundle = self[0];
+            bundle.secondaryFiles = [self[1]];
+            return bundle;
+          }
+      gtf_anno: gtf_anno
+    out: [arriba_pdf]
+
+  rsem:
+    run: ../tools/rsem_calc_expression.cwl
+    in:
+      bam: star_2-7-10a/transcriptome_bam_out
+      paired_end: paired_end
+      estimate_rspd: estimate_rspd
+      genomeDir: RSEMgenome
+      outFileNamePrefix: output_basename
+      strandedness: strand_parse/rsem_std
+    out: [gene_out, isoform_out]
+
+  rna_seqc:
+    run: ../tools/rnaseqc_2.4.2.cwl
+    in:
+      Aligned_sorted_bam: samtools_sort/sorted_bam
+      collapsed_gtf: RNAseQC_GTF
+      stranded: strand_parse/rnaseqc_std
+      unpaired:
+        source: rmats_read_type
+        valueFrom: |
+          $(self == "single" ? true : false)
```

Comment:
If true, return true...

Reply:
I think the intent was, if the read type is single, then return true (it's unpaired) else, return false. Do I have it wrong here?

Reply:
Yeah I'm just saying `self == "single"` is a boolean. There's no need to return anything else. This bit of code is basically:
```
if (true) { return true } elif (false) { return false }
```

Reply:
Ah, this js syntax is still a little new for me, so I guess there is sort of an implicit return and:
`$(self == "single" : false)` means the same thing?